### PR TITLE
O3 3685: Enable Deleting or Clearing Files in File Picker Component in Edit Mode

### PR DESCRIPTION
**Pull request by TrevorAntony**
## Requirements
Enable Deleting or Clearing Files in File Picker Component in Edit Mode

## Summary
When a file is selected using the file picker component and the form is later opened in edit mode, there is now an option to delete or clear the selected file. 

## Screenshots
Screenshot 2024-09-11 at 13 16 23
Screenshot 2024-09-11 at 13 16 31

## Related Issue
https://openmrs.atlassian.net/browse/O3-3685

### CODE REVIEW

File: src/components/inputs/file/file.component.tsx [hunk 1]
```diff
@@ -49,6 +49,13 @@ const File: React.FC<FormFieldInputProps> = ({ field, value, setFieldValue }) =>
     },
     [setFieldValue],
   );
+  const [clearedFiles, setClearedFiles] = useState([]);
```
Comment (samuelmale) [inline src/components/inputs/file/file.component.tsx:52]: Can you move this state declaration at the top? 

Question: Are you using the "clearedFiles" state at any point?

File: src/components/inputs/file/file.component.tsx [hunk 2]
```diff
@@ -49,6 +49,13 @@ const File: React.FC<FormFieldInputProps> = ({ field, value, setFieldValue }) =>
...
+  const handleClearFile = (fileId) => {
+    setClearedFiles((prevClearedFiles) => [...prevClearedFiles, fileId]);
+    setFieldValue(null);
+    setImagePreview(null);
+    setDataSource(null);
+  };
```
Comment (samuelmale) [inline src/components/inputs/file/file.component.tsx:58]: Can you wrap this handler wrapped in a useCallback?

**Review by samuelmale — CHANGES_REQUESTED** [review]
Thanks for working on this @TrevorAntony; Just a few general concerns:

1. Is the "delete" operation persistent on form submission? Do we void the complex obs?
2. Can you add unit test(s) to assert that the cleared obs gets voided?

File: src/components/inputs/file/file.component.tsx (hunk 2 above)
Comment (arodidev) [reply src/components/inputs/file/file.component.tsx:58]: @TrevorAntony I don't think we need to maintain the state of the cleared values, resetting the other three should suffice.

```ts
const handleClearFile = useCallback(() => {
  setFieldValue(null);
  setImagePreview(null);
  setDataSource(null);
}, []);
```

File: src/components/inputs/file/file.component.tsx (hunk 1 above)
Comment (TrevorAntony) [reply src/components/inputs/file/file.component.tsx]: noted

File: src/components/inputs/file/file.component.tsx (hunk 2 above)
Comment (TrevorAntony) [reply src/components/inputs/file/file.component.tsx]: yes done this

**Comment by TrevorAntony** [comment]
Having an issue where even after voiding/clearing the obs and successful form submission, when you relaunch the form in  edit mode, the cleared image/file is still displayed

**Pull request closed without merging**